### PR TITLE
Add Base64 missing tests and documentation fixes

### PR DESCRIPTION
**Pull request by jkbkupczyk**
- Adds missing tests for Base64 decoding and encoding (example Base64 data comes from [Base64 - Wikipedia](https://en.wikipedia.org/wiki/Base64) and [RFC - 4648](https://datatracker.ietf.org/doc/html/rfc4648))
- Updates Base64 documentation to reflect code and fixes some typos 

FYI @garydgregory @kinow

### CODE REVIEW

File: src/test/java/org/apache/commons/net/util/Base64Test.java [hunk 1]
```diff
@@ -208,9 +241,10 @@ public void testEncodeObject() {
     }
 
     @Test
-    @Ignore
     public void testEncodeToString() {
-        fail("Not yet implemented");
+        final Base64 base64 = new Base64();
+        final byte[] bytesToEncode = new byte[] {'l', 'i', 'g', 'h', 't', ' ', 'w', 'o', 'r'};
```
Comment (kinow) [inline src/test/java/org/apache/commons/net/util/Base64Test.java:246]: Random question here; was the example intended to be "light work" encoded? :slightly_smiling_face: Asking as it was used in the other tests, but doesn't really matter as this test is working fine :+1:

File: src/main/java/org/apache/commons/net/util/Base64.java [hunk 2]
```diff
@@ -603,10 +603,10 @@ int avail() {
     }
 
     /**
-     * Decodes a byte[] containing characters in the Base64 alphabet.
+     * Decodes a byte array containing characters in the Base64 alphabet.
      *
      * @param pArray A byte array containing Base64 character data
-     * @return a byte array containing binary data
+     * @return a byte array containing binary data; will return {@code null} if provided byte array is null.
```
Comment (kinow) [inline src/main/java/org/apache/commons/net/util/Base64.java:609]: s/is null/is {@code null}

File: src/main/java/org/apache/commons/net/util/Base64.java [hunk 3]
```diff
@@ -114,10 +114,10 @@ public class Base64 {
     // some state be preserved between calls of encode() and decode().
 
     /**
-     * Tests a given byte array to see if it contains only valid characters within the Base64 alphabet.
+     * Tests a given byte array to see if it contains any valid character within the Base64 alphabet.
```
Comment (kinow) [inline src/main/java/org/apache/commons/net/util/Base64.java:117]: Oh, I had to read the code to confirm it. To me the updated text looks correct, great catch!

File: src/test/java/org/apache/commons/net/util/Base64Test.java (hunk 1 above)
Comment (jkbkupczyk) [reply src/test/java/org/apache/commons/net/util/Base64Test.java:246]: nope, just a random example from [Base64 - Output Padding](https://en.wikipedia.org/wiki/Base64#Output_padding) page in wikipedia 😀

File: src/main/java/org/apache/commons/net/util/Base64.java (hunk 3 above)
Comment (jkbkupczyk) [reply src/main/java/org/apache/commons/net/util/Base64.java:117]: thanks 😊

File: src/main/java/org/apache/commons/net/util/Base64.java (hunk 2 above)
Comment (jkbkupczyk) [reply src/main/java/org/apache/commons/net/util/Base64.java]: fixed in 7c11889a03a3754a9abd6874cf8b2d06707d0cb9

**Review by kinow — APPROVED** [review]
Looks good to me! Commits need to be squashed but we can take care of that later. Thanks @jkbkupczyk !

File: src/test/java/org/apache/commons/net/util/Base64Test.java (hunk 1 above)
Comment (garydgregory) [reply src/test/java/org/apache/commons/net/util/Base64Test.java:246]: > Random question here; was the example intended to be "light work" encoded? 🙂 Asking as it was used in the other tests, but doesn't really matter as this test is working fine 👍

In the future, we could split up test method per test source, like `testWikipediaExamplesAbc`, `testRfc123FeatureX`, and so on.

**Pull request merged**